nectar/sample-config.toml

```diff
@@ -23,7 +23,13 @@ dai = 1000
 [maker.maximum_possible_fee]
 # An estimation of the maximum fee that we would expect to pay, used to ensure we always have enough
 # balance to execute an order we publish.
-bitcoin = 0.00009275 
+bitcoin = 0.00009275
+
+# Strategies used for Bitcoin fee handling.
+[maker.fee_strategies.bitcoin]
+# A static sat/byte value can be specified or you can relay on bitcoind's estimatesmartfee feature.
+strategy = "static"
```

Comment:
I assume the keys (like `sats_per_byte`) vary based on this field? You could elevate this to a toml table and let that one control the keys:

```toml
[maker.fee_strategies.bitcoin.static]
sats_per_byte = 10

[maker.fee_strategies.bitcoin.bitcoind_estimate]
mode = "conservative"
```

Reply:
This should represent nicely as an enum in Rust and hence guarantee you at compile time that you don't get a "mode" if the user selected "static" and vice versa.

Reply:
> This should represent nicely as an enum in Rust and hence guarantee you at compile time that you don't get a "mode" if the user selected "static" and vice versa.

I tried and that is how I spent most of my Friday. The toml crate does not seem to allow that.
However I only tried an enum with an anonymous field (newtype) not with a struct.

I don't really want to fight again with the crate at this stage. Happy to either review at a later stage or for you to give it a go.

Reply:
I think I remember doing something similar in cnd so I ought it would work. Happy to give it a go in a follow-up :)